packages/mdc-button/mdc-button.scss

```diff
@@ -87,110 +57,104 @@
     border: 0;
   }
 
-  &--dense {
-    height: 32px;
-    font-size: .8125rem; // 13sp
-    line-height: 32px;
+  &:not(.mdc-ripple-upgraded) {
+    -webkit-tap-highlight-color: rgba(98, 25, 130, .3);
   }
 
-  // TODO(cristobalchao): Disable ink wash on hover and alter elevation instead for raised surfaces.
-  &--raised {
-    @include mdc-elevation(2);
-    @include mdc-elevation-transition;
-
-    min-width: 88px;
-
-    &:active {
-      @include mdc-elevation(8);
-    }
-
-    @each $theme-style in (primary, accent) {
-      &.mdc-button--#{$theme-style} {
-        @include mdc-ripple-base;
-        @include mdc-ripple-bg((pseudo: "::before", base-color: white, opacity: .14));
-        @include mdc-ripple-fg((pseudo: "::after", base-color: white, opacity: .14));
-      }
-    }
+  fieldset:disabled &,
+  &:disabled {
+    color: rgba(98, 25, 130, .38);
+    cursor: default;
+    pointer-events: none;
 
     @include mdc-theme-dark(".mdc-button") {
-      @include mdc-theme-prop(background-color, primary);
-      @include mdc-theme-prop(color, text-primary-on-primary);
-
-      // postcss-bem-linter: ignore
-      &::before {
-        // We are explicitly not fully adhering to Material Design here.
-        // This should be the 700-shade when active instead of a black shade.
-        // Due to the complexity involved in adhering fully it is being ignored.
-        // Instead re-using the existing architecture for shading works just fine.
-        color: black;
-      }
+      @include mdc-theme-prop(color, text-disabled-on-dark);
     }
   }
 
-  &--primary {
-    @include mdc-theme-prop(color, primary);
-
-    @include mdc-theme-dark(".mdc-button") {
-      @include mdc-theme-prop(color, primary);
-    }
-
-    // postcss-bem-linter: ignore
-    &.mdc-button--raised {
-      @include mdc-theme-prop(background-color, primary);
-      @include mdc-theme-prop(color, text-primary-on-primary);
+  @include mdc-theme-dark(".mdc-button") {
+    @include mdc-ripple-base;
+    @include mdc-ripple-bg((pseudo: "::before", base-color: white, opacity: .16));
+    @include mdc-ripple-fg((pseudo: "::after", base-color: white, opacity: .16));
+    @include mdc-theme-prop(color, text-primary-on-dark);
 
-      // postcss-bem-linter: ignore
-      &::before {
-        color: black;
-      }
+    &:not(.mdc-ripple-upgraded) {
+      -webkit-tap-highlight-color: rgba(white, .18);
     }
   }
+}
 
-  &--accent {
-    @include mdc-theme-prop(color, accent);
-
-    @include mdc-theme-dark(".mdc-button") {
-      @include mdc-theme-prop(color, accent);
-    }
+.mdc-button--raised {
```

Comment:
It has been that we put &__ to the element contained within the component, but not on modifier class. For example: `mdc-checkbox--disabled`, `mdc-form-field--align-end`, `mdc-toolbar--flexible` but also this is not a enforced style.